src/user.py

```diff
@@ -16,8 +20,17 @@ class User(BaseModel):
     """
 
     username: str
-    admin: bool
-    password: str
+    admin: str | bool
+    password: str = Field(False)
+
+    def __init__(self, **data: dict[str, typing.Any]) -> None:
+        """Initialize a new User instance.
+
+        Parameters:
+            data: A dictionary containing the user data.
+        """
+        super().__init__(**data)
+        self._set_password()
```

Comment:
It would be better to do something like `self.password = _generate_password()` where `_generate_password` can just be a function at the module level since it probably doesn't need access to the user data. This makes it clearer what is happening rather than initialising instance attributes in methods which should generally be avoided

Reply:
Agree, changed.